gprofiler/merge.py

```diff
@@ -332,6 +333,41 @@ def _enrich_and_finalize_stack(
     return f"{enrich_data.application_prefix}{enrich_data.container_prefix}{stack} {count}"
 
 
+def concatenate_from_external_file(
+    collapsed_file_path: str,
+    obtained_metadata: Metadata,
+) -> Tuple[str, str, str]:
+    """
+    Concatenate all stacks from all stack mappings in process_profiles.
+    Add "profile metadata" and metrics as the first line of the resulting collapsed file.
+    """
+
+    lines = []
+    start_time = None
+    end_time = None
+
+    # TODO: container names and application metadata
+    with open(collapsed_file_path) as file:
+        for index, line in enumerate(file):
+            if index == 0:
+                read_metadata = json.loads(line[1:])
+                metadata = merge_dicts(read_metadata, obtained_metadata)
+                try:
+                    start_time = parse_iso8601_timestamp(metadata["start_time"])
+                    end_time = parse_iso8601_timestamp(metadata["end_time"])
+                except KeyError:
+                    pass
+                try:
+                    metadata["run_arguments"]["func"] = str(metadata["run_arguments"]["func"])
```

Comment:
I suppose you're casing it here to replace the "function" object with a string?

In any case, we don't need it, so you can just ensure `func` is popped out from `run_arguments` (we pass `args.__dict__` to `GProfiler` class, you can pass a copy of the dict with `func` popped). Please explain in comment why it's done, because this code isn't clear

Reply:
You are right, the purpose was to omit json errors when function object was encountered. I'll pop it from run arguments then.